### ISSUE TITLE
Cannot invoke simple serverless function

### ISSUE THREAD

**Issue by atsag** [serverless/serverless-azure-functions#128]
I created a new instance of a serverless function based on the azure-nodejs template, initialized according to the steps defined at [the documentation](https://serverless.com/framework/docs/providers/azure/guide/quick-start/) . I managed to successfully upload it to Azure (using either a custom service principal or a login-created service principal), however upon invoking the function, an error message appears with the following execution trace:

```Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Invoke invoke
Serverless: Logging in to Azure
 
  Error --------------------------------------------------
 
  [object Object]
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Stack Trace --------------------------------------------
 
Error: [object Object]
    at module.exports.logError (C:\Users\user\AppData\Roaming\npm\node_modules\serverless\lib\classes\Error.js:92:11)
    at initializeErrorReporter.then.catch.e (C:\Users\user\AppData\Roaming\npm\node_modules\serverless\bin\serverless:62:3)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at C:\Users\user\AppData\Roaming\npm\node_modules\serverless\bin\serverless:60:9
    at Object.<anonymous> (C:\Users\user\AppData\Roaming\npm\node_modules\serverless\bin\serverless:63:4)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     win32
     Node Version:           8.12.0
     Serverless Version:     1.32.0
```

However, if the changes introduced with [this commit](https://github.com/serverless/serverless-azure-functions/commit/ff50c6cc35d2705d6a90c954ab727ae26359320f?diff=split) to  invoke/azureInvoke.js are reverted, I can successfully invoke the sample hello function.

**Comment by pankajku** [comment]
Facing the same issue.

**Comment by nickwesselman** [comment]
Same. Is the Azure Functions plugin even being maintained?

**Comment by pietroiusti** [comment]
Same, any workarounds?

**Comment by JohnMilazzo** [comment]
Same issue here.

**Comment by iceskel** [comment]
Still an issue (tried on macOS mojave). ☹️

**Comment by UtkarshYeolekar** [comment]
I am also facing the same issue, simple functions are not getting invoked. Also, if i tried to pass the json with the data param. It is giving me the error that
"The specified input data isn't a valid JSON string. Please correct it and try invoking the function again." Event though i am providing a valid json.

**Comment by dontmatta** [comment]
Here's how I got it to work.  I changed my package.json file:

`  "devDependencies": {
    "serverless-azure-functions": "*"
  }`
to
`  "dependencies": {
    "serverless-azure-functions": "^0.2.0"
  }`

There's probably a better way to do this but it worked for me.

**Comment by pfeilbr** [comment]
> Here's how I got it to work. I changed my package.json file:
> 
> ` "devDependencies": { "serverless-azure-functions": "*" }`
> to
> ` "dependencies": { "serverless-azure-functions": "^0.2.0" }`
> 
> There's probably a better way to do this but it worked for me.

This worked for me.

**Comment by callmekatootie** [comment]
**tl;dr** Run the following command:

`$ az functionapp config appsettings set --name <your_function_app_name> --resource-group <your_resource_group> --settings "AzureWebJobsSecretStorageType=Files`

and then try to invoke. It should work.

**Long version:**
The `[object Object]` error message, when debugged, gives the following:

```
{
    "Message": "An error has occurred.",
    "ExceptionMessage": "Runtime keys are stored on blob storage. This API doesn't support this configuration. Please change Environment variable AzureWebJobsSecretStorageType value to 'Files'. For more info, visit https://aka.ms/funcsecrets",
    "ExceptionType": "System.InvalidOperationException",
    "StackTrace": "   at Kudu.Core.Functions.FunctionManager.<GetKeyObjectFromFile>d__9`1.MoveNext() in C:\\Kudu Files\\Private\\src\\master\\Kudu.Core\\Functions\\FunctionManager.cs:line 141\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Kudu.Core.Functions.FunctionManager.<GetMasterKeyAsync>d__11.MoveNext() in C:\\Kudu Files\\Private\\src\\master\\Kudu.Core\\Functions\\FunctionManager.cs:line 213\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Kudu.Services.Functions.FunctionController.<GetMasterKey>d__11.MoveNext() in C:\\Kudu Files\\Private\\src\\master\\Kudu.Services\\Functions\\FunctionController.cs:line 121"
}
```

which lead me to this [wiki article](https://github.com/Azure/azure-functions-host/wiki/Changes-to-Key-Management-in-Functions-V2) stating that Azure Function used to have File system for key storage mechanism but they have now started using Blob Storage instead.

I am guessing this plugin requires the File system and not Blob Storage, and thus the error.

If you run the following command, we instruct Azure Function to use File Storage itself, thus resolving the issue:

`$ az functionapp config appsettings set --name <your_function_app_name> --resource-group <your_resource_group> --settings "AzureWebJobsSecretStorageType=Files`
